conda/gateways/subprocess.py

```diff
@@ -101,15 +101,12 @@ def subprocess_call(
         stdout=pipe,
         stderr=pipe,
         env=env,
+        text=True,  # open streams in text mode so that we don't have to decode
     )
     ACTIVE_SUBPROCESSES.add(process)
 
     # decode output, if not PIPE, stdout/stderr will be None
     stdout, stderr = process.communicate(input=stdin)
```

Comment:
What if `stdin` is `bytes`? Do we need to decode that one?

Reply:
Comparing to the above similar `any_subprocess` function that retains the hasattr(stdout, "decode")... code, should it match this updated version?

Can we use Python 3's builtin subprocess module "preferred API" instead?

Reply:
@dholth probably